site/app/controllers/MiscController.php

```diff
@@ -83,6 +83,12 @@ private function checkValidAccess($is_zip) {
             // gradeable to get temporary info from
             // if team, get one of the user ids via the team id
             $current_gradeable = $this->core->getQueries()->getGradeable($path_gradeable_id, $current_user_id);
+            if($current_gradeable->getPeerGrading()) {
+                $peer_grade_set = $this->core->getQueries()->getPeerAssignment($path_gradeable_id, $current_user_id);
+                if(in_array($path_user_id, $peer_grade_set)) {
+                    return true;
+                }
+            }
```

Comment:
peer's should only be able to look at the files of a student they are grading.  I took away the download button on the grading page, to deter that attempt, but you're right, we should have that check explicitly in the code